### DISCUSSION TITLE
A Radical Market Approach towards FIP Consensus

### DISCUSSION THREAD

**Discussion by Fatman13** [filecoin-project/FIPs#1040] [Enhancements - Technical]
## Introduction

Filecoin has been using the [soft consensus](https://github.com/filecoin-project/FIPs/discussions/844#discussioncomment-7322514) process for accepting or rejecting FIP proposal since its mainnet, which has its merits and flaws. This thread intends to solicit discussion on using a radical market approach to improve current FIP consensus process, which is inspired by [Radical Markets](https://www.amazon.com/Radical-Markets-Uprooting-Capitalism-Democracy/dp/0691177503) authored by Eric Posner and Glen Weyl.

### Why Market Approach?

For lacking other competent alternatives, markets have been the most efficient way to capture participants' preferences since emergence of land and labor markets during industrial revolution. Like in real world there is no political market to reflect constituency's preferences in different policies, there is no market in Filecoin consensus process to reflect network stakeholder's preferences in the direction of where the network should be heading. It is thus crucial to establish such market so that the community could collectively steer the network to its betterment.

### An Example Market

An example describing how a consensus market that could proximate individual's preferences given a range of choices could be found [here](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf). It is recommended that readers of this FIP discussion to go through the example in the aforementioned link from page 2 to 3.

### Key Takeaway

There are key elements for a consensus market to work, on which the proposed market approach relies heavily.

- Accumulation of voting credits over time
- A KYC scheme or a close alternative
- A quadratic voting scheme 
- A wide selection of issues to vote on

## Proposed Market Approach

*Each aspect of the market approach below is subject to better alternative solution and further fine tuning per discussion.*

###  Acquiring Voting Credits

A staking mechanism to allow network stakeholder, ie token holders, to accumulate voting credits (vfil) over time. The amount of voting credits one is awarded over a time interval could be a function of amount of token staked and the time it is committed in staking. And once the time commitment is fulfilled, the amount of fil token would then be released back to the staker. For example... 

```
amount of vfil awarded daily = f( amount of fil staked, amount of time to be staked )
```

The staking time could be [180, 1080] days in conformity with sector lifetime defined in [core protocol](https://spec.filecoin.io/#section-algorithms.cryptoecon.initial-parameter-recommendation) so that token staker and miner could both use the same vfil vesting function.

Additionally, vfil vesting function could take into consideration that the longer tokens are to be staked the more vfil it could reward the staker to incentivize long term alignment which helps alleviating potential problems of voting out of short term interests.

### Voting Scope

All FIPs are to be voted for acceptance or rejection to capture community's preferences on where the network will be heading. The larger the scope of the voting is, the better it would reflect preferences.

> In general, we believe that, with appropriate regulations, the broader the scope of use of voice credits, the better.
> -- [excerpts](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) from page 34

### QV

Quadratic voting has been used by Filecoin in projects like network public goods and regtroPGF. QV concept works as the following...

> the voice credits are used to cast votes according to a quadratic or square root function. One voice credit buys one vote on a given issue; 4 buy two votes; 400 buy only 20 votes; and so on.
> -- [excerpts](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) from page 3

QV outperforms traditional 1 person 1 vote (1p1v) system in...

> QV achieves a perfect balance between the free-rider and the tyranny of the majority problems.
> -- [excerpts](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) from page 22

A fundamental issue that QV tries to address is that...

> ensure that no one’s happiness can be increased without decreasing anyone else’s, a condition called Pareto efficiency, and that the total happiness is distributed fairly.
> -- [excerpts](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) from page 23

More on "why quadratic voting" can be found under "Quadratic Voting" section on page 19 of this [link](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf).

### KYC

A key component for this voting scheme to work is KYC which is not mentioned in the [book](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) as it is always assumed that KYC just works in real world, but in Web3 it would be much more challenging to properly KYC on a global scale. It could be either a combination of Web2 accounts (Github, Twitter, Slack), or maybe a minimum staking threshold say the amount of pledge required for median node QAP among all mining nodes. Any suggestion on KYC or alternatives is welcomed.

### Voting & Acceptance Criteria 

TBD

### QV

QV is not without its limitations. For example...

> similar problem may arise from collusion, vote-buying, or fraud, just as in 1p1v systems.
> -- [excerpts](https://publics22.classes.ryansafner.com/readings/Posner-Weyl-2017-Ch2.pdf) from page 23

vfil tokens could be soul bound to reduce the risk of bribing. For example, stakers transfers its vfil to others to vote for a price.

### KYC

A flawed KYC or alternative may result in DDOS attacks wrt voting. For example, token holder distribute its staking so that it takes advantage of using 1 vfil to buy one vote and never go up to higher cost of acquiring multiple number of votes.

### Enforcement

Quick reactions and strong stance on frauds maybe challenging to achieve on large scale. For example,  it would be very damaging to voting's credibility if no action were to be carried out investigating and mitigating fraud. 

<img width="32" height="32" alt="mfga-punks" src="https://github.com/user-attachments/assets/d9964530-13cd-4653-a1e0-3f3399bc6fcd" />

*A proposal by [Ancients Research](https://ancientsresearch.xyz/)*

**Comment by The-Wayvy** [comment]
Soft Consensus = Juan supports this FIP

**Comment by 15012700225** [comment]
Greate FIP, the cornerstone for moving forward. Support this.
